### DISCUSSION TITLE
armv7 docker container

### DISCUSSION THREAD

**Discussion by Sonusss** [music-assistant/support#1272] [Feature requests and ideas]
Will it be possible to get an armv7 (Raspberry 32bits) docker container ?

**Comment by marcelveldt** [comment]
Sorry but no, its 2023 so 64bits is the golden standard nowadays.

Still having to publish 32 bits makes building way slower and more difficult as we're already seeing some dependencies not providing 32 bit builds anymore.

**Comment by Sonusss** [comment]
Make sense, however regarding arm platform many apps and servers are still in the migration path, just because 64bits is not mandatory for most of them... Do not follow Microsoft policy...

**Reply by marcelveldt** [reply]
Yeah I know but MA is considered an advanced project which is best suited on more powerful devices, so 2GB of memory as absolute minimum but more is better for large libraries. So, the target for 64 bits anyways.

Also having to support armv7 in the docker builds alone will extend building times with a couple of hours for difficult dependencies so too much of a penalty for just a few users still running 32bits OS.

I understand that is not so nice to hear if you're still running 32 bits OS but if you are I'm pretty sure you have the upgrade to 64bits already somewhere on your TODO list ;-)

**Comment by riesi77** [comment]
I'd like to underline the request for armv7. Migrating my raspi to 64 bit is a big project, because I have many different applications running (including data). And I think 32 bit is still the majority on raspis, so  I'm not alone on that. I didn't even consider to migrate until now.
Do you have download numbers that give an overview on which version was downloaded how often?

**Reply by marcelveldt** [reply]
I think this chart says it all ?
https://analytics.home-assistant.io/

Only a very small amount of pi4 users still running 32bits.

Note that a pi3 is not powerful enough to run MA (and HA even) so I dont account for that

**Reply by riesi77** [reply]
That's a big surprise - and a very valuable statistics page. Thanks for taking the effort. I will put my 64 bit migration on top of my list ;-).

**Comment by Sonusss** [comment]
I would like to give a try building my own docker image for armv7.
Can someone help ?

**Reply by marcelveldt** [reply]
What is holding you back from running 64 bits ? Do note that a pi 3 is not powerful enough to run both HA and MA, so a pi4 with 4gb is recomended, 2gb the absolute minimum if you have a small setup.

**Reply by Sonusss** [reply]
It is a raspberry pi4 with 4GB memory but it runs other software not available as 64bits version. These are pure software not docker images.
I thought then that it would be easyer to adapt one component to 32bits than recompiling all other stuff to 64bits version...

**Reply by marcelveldt** [reply]
Well, last time I checked the image did not want to build for armv7 or it was terribly slow so I ditched it due to the fact only a very small amount of users still runs 32bits. Feels kind of a waste to slowdown the build process to 90 minutes (from 10 minutes) by enabling armv7 support. That said, I want to switch to the alpine images from HA now that they migrated to python 3.11 so that might fix the issue as the HA project has prebuilt musl wheels for all dependencies.

**Reply by Sonusss** [reply]
As it looks not straightforward, I'll wait or try to migrate my stuff. As you explained this is the way to follow...
Many thanks for your kind attention

**Reply by Sonusss** [reply]
I hosted a docker image on my own computer (Ubuntu 20.04 LTS) in the course to give a real try to Music Assistant. This is why you can see a new issue posted by me ... https://github.com/music-assistant/hass-music-assistant/issues/1315
